flake8_trio.py

```diff
@@ -290,18 +280,48 @@ def visit_Import(self, node: ast.Import):
 
     # ---- 110 ----
     def visit_While(self, node: ast.While):
-        self.check_for_110(node)
+        self.check_for_trio110(node)
         self.generic_visit(node)
 
-    def check_for_110(self, node: ast.While):
+    def check_for_trio110(self, node: ast.While):
         if (
             len(node.body) == 1
             and isinstance(node.body[0], ast.Expr)
             and isinstance(node.body[0].value, ast.Await)
-            and get_trio_scope(node.body[0].value.value, "sleep", "sleep_until")
+            and get_matching_call(node.body[0].value.value, "sleep", "sleep_until")
         ):
             self.error("TRIO110", node)
 
+    # if with has a withitem `trio.open_nursery() as <X>`,
+    # and the body is only a single expression <X>.start[_soon](),
+    # and does not pass <X> as a parameter to the expression
+    def check_for_trio112(self, node: Union[ast.With, ast.AsyncWith]):
+        # body is single expression
+        if len(node.body) != 1 or not isinstance(node.body[0], ast.Expr):
+            return
+        for item in node.items:
+            # get variable name <X>
+            if not isinstance(item.optional_vars, ast.Name):
+                continue
+            var_name = item.optional_vars.id
+
+            # check for trio.open_nursery
+            nursery = get_matching_call(item.context_expr, "open_nursery")
+
+            # isinstance(..., ast.Call) is done in get_matching_call
+            body_call = cast(ast.Call, node.body[0].value)
```

Comment:
I'd use an `assert` rather than a `cast()` here.

Reply:
`assert` is actually not possible here, the isinstance call is done later inside `get_matching_call` (which if it inside there evaluates false propagates to the condition calling it being false).

the easiest alternative is a redundant `isinstance` in the condition

I suppose the cast is incorrect in that `body_call` isn't actually a `Call` *yet*, but is for brevity/convenience later.